File: designPattern/Composite/Basic/src/Entry.java
public abstract class Entry{
    public abstract String getName();
    public abstract int getSize();

    public void printLine(){
        printLine("");
    }
    public abstract void printLine(String prefix);

    public abstract Entry add(Entry entry);

    public String toString(){
        return "/"+getName()+" ("+getSize()+")";
    }
}

File: designPattern/Composite/basicPython/Entry.py
import abc


class Entry:
    __metaclass__ = abc.ABCMeta

    @abc.abstractmethod
    def getName(self):
        pass

    @abc.abstractmethod
    def getSize(self):
        pass

    def execute(self, prefix):
        self.print_line(prefix)

    @abc.abstractmethod
    def print_line(self, prefix):
        pass

    def getInformation(self):
        return str(self.getName()) + " ("+str(self.getSize())+")"
